src/sql/util.py

```diff
@@ -527,3 +528,33 @@ def validate_mutually_exclusive_args(arg_names, args):
             f"{pretty_print(specified_args)} are specified. "
             "You can only specify one of them."
         )
+
+
+def validate_nonidentifier_connection(args, command):
+    """
+    Raises UsageError if a connection is passed to `%sql/%%sql` through
+    object property, list, or dictionary.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Contains %sql command line arguments values
+    command : sql.command.SQLCommand
+        Contains sql query after variable expansion if necessary
+    """
+    command_sql_first_line = command.sql.split("\n")[0].split(" ")
+    if (
+        len(command_sql_first_line) == 1
+        and bool(command_sql_first_line[0])
+        and not command_sql_first_line[0].isidentifier()
+        and not (args.persist_replace or args.persist or args.append)
+    ):
```

Comment:
this logic is hard to read

`command_sql_first_line` isn't the first line, these are the first line tokens 

I think if you move this to wherever we're parsing the existing logic, you'll find similar logic already, and it'll simplify this

Reply:
Only test `isidentifier()` and `is_valid_python_code` in `validate_nonidentifier_connection` : [fixed](https://github.com/ploomber/jupysql/pull/821/files#diff-653c15e6161c97f356be04e072838ed15014af275c06019a7692ab727fa44b97R543)

Simplified if conditions by using what's available in `command.py` : [fixed](https://github.com/ploomber/jupysql/pull/821/files#diff-769097f5e36be57ddec10c78df0900db8c1e254f5485bc94367aa7fcb07991c5R65-R69)